Sources/TuistCache/Cache/FileArchiver.swift

```diff
@@ -0,0 +1,15 @@
+import Foundation
+import TSCBasic
+import Zip
+
+protocol FileArchiving {
+    func zip(xcframeworkPath: AbsolutePath, hash: String) throws -> AbsolutePath
+}
+
+class FileArchiver: FileArchiving {
+    func zip(xcframeworkPath: AbsolutePath, hash: String) throws -> AbsolutePath {
```

Comment:
Considering zipping a `.xcframework` is a temporary action required to upload the framework, and that afterward the .xcframework is not necessary. Would it make sense to change this method to:

- Generate the .zip in a temporary directory.
- Provide the consumer of the method with a mechanism for deleting it.

What about something along the lines of:

```swift
fund zip(xcframeworkPath: AbsolutePath) throws -> (path: AbsolutePath, delete: () throws -> Void)
```

Notice that we are not passing the hash because it's not necessary that the zipped xcframework in the temporary directory contains that name. The backend will be responsible for storing it with the right name.

Reply:
I think we can simplify this even further by doing:

```swift
class ZipFile {
  let path: AbsolutePath
  deinit {
    // Delete the temporary directory that contains the file
  }
}

fun zip(xcframeworkPath: AbsolutePath) throws -> (path: AbsolutePath)
```

That way the file gets deleted when the consumer of the zip file releases it from memory.

Reply:
Implemented something like your last idea here 🔝